src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1525,8 +1525,15 @@ def update_stack_component(
                         "Service connector with ID "
                         f"{component_update.connector} not found."
                     )
+            else:
+                existing_component.connector = None
+                existing_component.connector_resource_id = None
 
             if service_connector:
+                logger.debug(
+                    "Updating component %s with connector %s",
+                    existing_component,
+                )
                 existing_component.connector = service_connector
```

Comment:
I'd suggest to move this entire block up into the `if component_update.connector:` block after L1521

Reply:
I think it's fine as it is, because this immediately fixes the UI as well. There is no disconnect field in the component update model and we shouldn't add one. The alternative also requires dashboard changes.